### PR TITLE
FINERACT-1981: Enhance EMI Calculator Daily Interest calculation

### PR DESCRIPTION
**Pull request by janez89**
This P.R. enhances the EMI Calculator Daily Interest Calculation.

- Includes EMI Calculator Tests
- getPayableDetails to calculate Daily Interest
- Fixed Interest Refund tests

### CODE REVIEW

File: fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/data/PayableDetails.java [hunk 1]
```diff
@@ -27,5 +27,6 @@ public class PayableDetails {
     private final Money emi;
     private final Money payablePrincipal;
     private final Money payableInterest;
+    private final Money accrualInterest;
```
Comment (adamsaghy) [inline fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/data/PayableDetails.java]: What does this field represent?

File: fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java [hunk 2]
```diff
@@ -175,6 +181,10 @@ public PayableDetails getPayableDetails(final ProgressiveLoanInterestScheduleMod
             interestPeriod = repaymentPeriod.getInterestPeriods().stream()
                     .filter(ip -> targetDate.isAfter(ip.getFromDate()) && !targetDate.isAfter(ip.getDueDate())).findFirst().orElseThrow();
         }
+
+        // we should calculate accrual this part before adjust the interest period due date
+        final BigDecimal dailyAccrual = includeAccrual ? getOneDayAccrual(scheduleModelCopy, interestPeriod) : BigDecimal.ZERO;
```
Comment (adamsaghy) [inline fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java]: Its rather daily interest than accrual... :/

File: fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java [hunk 3]
```diff
@@ -188,13 +198,44 @@ public PayableDetails getPayableDetails(final ProgressiveLoanInterestScheduleMod
         calculateOutstandingBalance(scheduleModelCopy);
         calculateLastUnpaidRepaymentPeriodEMI(scheduleModelCopy);
 
+        // Accrual interest calculation
+        Money cumulatedAccrualInterest = scheduleModelCopy.getZero();
```
Comment (adamsaghy) [inline fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java]: I would rather keep it separately, not as part of payable details

**Review by kjozsa — APPROVED** [review]

**Comment by janez89** [comment]
We had a PoC about the feature request and now this is the final version.

**Review by galovics — APPROVED** [review]

**Pull request merged**